qiskit/visualization/text.py

```diff
@@ -1013,8 +1018,12 @@ def add_connected_gate(node, gates, layer, current_cons):
                     layer.set_qubit(node.qargs[i], gate)
                     current_cons.append((actual_index, gate))
 
-        if isinstance(op, Measure):
-            gate = MeasureFrom()
+        if isinstance(op, (Measure, MeasurePauli)):
+            if isinstance(op, Measure):
+                gate = MeasureFrom()
+            else:
+                basis = op.params.upper()
+                gate = MeasureFrom(basis)
```

Comment:
This seems like a very similar situation than here: https://github.com/Qiskit/qiskit-terra/pull/5311/files#r810540202

Reply:
yeap.. same situation (this might need a test)
```python
circuit = QuantumCircuit(6, 6)
circuit.measure_pauli("x", [0], [0])
circuit.barrier()
circuit.measure_pauli("y", [1], [1])
circuit.barrier()
circuit.measure_pauli("z", [2], [2])
circuit.barrier()
circuit.measure_pauli("xyz", [3, 4, 5], [3, 4, 5])
print(circuit)
```
```
     ┌─┐ ░     ░     ░    
q_0: ┤M├─░─────░─────░────
     └╥┘ ░ ┌─┐ ░     ░    
q_1: ─╫──░─┤M├─░─────░────
      ║  ░ └╥┘ ░ ┌─┐ ░    
q_2: ─╫──░──╫──░─┤M├─░────
      ║  ░  ║  ░ └╥┘ ░ ┌─┐
q_3: ─╫──░──╫──░──╫──░─┤M├
      ║  ░  ║  ░  ║  ░ └╥┘
q_4: ─╫──░──╫──░──╫──░──╫─
      ║  ░  ║  ░  ║  ░  ║ 
q_5: ─╫──░──╫──░──╫──░──╫─
      ║  ░  ║  ░  ║  ░  ║ 
c: 6/═╩═════╩═════╩═════╩═
      0     1     2     3 
```

So, same workaround?
```suggestion
        if isinstance(op, (Measure, MeasurePauli)):
            if isinstance(op, MeasurePauli):
                gate = MeasureFrom(''.join(op.params))
            elif isinstance(op, Measure):
                gate = MeasureFrom()
```